### DISCUSSION TITLE
Can We Use a Custom TLS to Connect to a Server in mitmproxy?

### DISCUSSION THREAD

**Discussion by AnsahMohammad** [mitmproxy/mitmproxy#7166] [Q&A] [answered]
Hi,
We're working on a project that involves connecting to a server using a custom TLS implementation. Could you guide us on how to manipulate the TLS connection with mitmproxy?
[We're using a custom fork of OpenSSL called OQS-OpenSSL].

**Comment by mhils** [comment, marked as answer]
Hi @AnsahMohammad,

> Could you guide us on how to manipulate the TLS connection with mitmproxy?

mitmproxy uses OpenSSL from [cryptography](https://cryptography.io/en/stable/). So you need to compile cryptography with your OpenSSL fork, and then mitmproxy should pick it up automatically.

> [We're using a custom fork of OpenSSL called OQS-OpenSSL].

I'm so sorry to hear that you have to go through that in your life, hope you are doing well otherwise. 🫂
